app/models/commit_status.rb

```diff
@@ -17,13 +17,47 @@ def status_list
   private
 
   def combined_status
-    @combined_status ||= begin
-      GITHUB.combined_status(@repo, @reference).to_h
-    rescue Octokit::NotFound
-      {
-        state: "failure",
-        statuses: [{"state": "Reference", description: "'#{@reference}' does not exist"}]
-      }
-    end
+    @combined_status ||= merge(github_status, release_status)
+  end
+
+  # simplistic merge that overrides state and combines messages
+  def merge(a, b)
+    return a unless b
+    a[:state] = b.fetch(:state)
+    (a[:statuses] ||= []).concat b.fetch(:statuses)
+    a
+  end
+
+  def github_status
+    GITHUB.combined_status(@stage.project.user_repo_part, @reference).to_h
+  rescue Octokit::NotFound
+    {
+      state: "failure",
+      statuses: [{"state": "Reference", description: "'#{@reference}' does not exist"}]
+    }
+  end
+
+  # checks if other stages that deploy to the same hosts as this stage have deployed a newer release
+  # optimized to sql instead of AR fanciness to make it go from 1s -> 0.01s on our worst case stage
+  # @return [nil, error-state]
+  def release_status
+    return unless DeployGroup.enabled?
+    return unless number = @reference[Release::VERSION_REGEX, 1]
+
+    deploy_groups = @stage.deploy_groups.reorder(nil).pluck(:deploy_group_id)
+    influencing_stages = DeployGroupsStage.reorder(nil).where(deploy_group_id: deploy_groups).pluck(:stage_id).uniq
+    last_deployed = Deploy.reorder(nil).where(stage_id: influencing_stages).group(:stage_id).pluck('max(id)')
+    last_deployed_references = Deploy.reorder(nil).where(id: last_deployed).pluck(:reference)
+    last_deployed_numbers = last_deployed_references.map { |r| r[Release::VERSION_REGEX, 1] }.compact
```

Comment:
extracted it into a method ... looks alright ?
 ... otherwise could extract it to stage ... but that feels weird ... (not fetching complete deploy groups for speed)